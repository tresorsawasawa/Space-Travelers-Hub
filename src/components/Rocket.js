import React from 'react';
import PropTypes from 'prop-types';

const Rocket = ({
  id, description, name, image,
}) => (
  <div className="rocket-container flex-center-center">
    <div className="image-container">
      <img src={image} alt={name} />
    </div>
    <div className="rocket-infos">
      <h3 className="rocket-name">{name}</h3>
      <p className="description">{description}</p>
      <button type="button" id={id} className="reserve-btn">Reserve Rocket</button>
    </div>
  </div>
);

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Rocket;
