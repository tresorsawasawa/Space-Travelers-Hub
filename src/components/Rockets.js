import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getRocketsDispatcher } from '../redux/rockets/rockets';
import Rocket from './Rocket';

const Rockets = () => {
  const dispatch = useDispatch();
  const rocketsData = useSelector((state) => state.rocketsReducer);

  useEffect(() => {
    if (!rocketsData.isDataStored) {
      dispatch(getRocketsDispatcher());
    }
  }, []);

  const rockets = rocketsData.data;
  let rocketsList = [];
  if (rockets) {
    rocketsList = rockets.map((rocket) => (
      <li key={rocket.id} className="rocket-item">
        <Rocket
          className="rocket"
          description={rocket.description}
          image={rocket.image}
          name={rocket.rocket_name}
          id={rocket.id}
          reserved={rocket.reserved}
        />
      </li>
    ));
  }

  return (
    <section>
      <div className="rockets">
        <ul className="rockets-list">
          {rocketsList}
        </ul>
      </div>
    </section>
  );
};

export default Rockets;
