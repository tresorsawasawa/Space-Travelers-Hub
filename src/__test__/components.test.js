import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Rockets from '../components/Rockets/Rockets';
import Rocket from '../components/Rockets/Rocket';

it('Test if Rocket render correctly', () => {
  const rocket = {
    id: 1,
    description: 'This is the Falcon1 Rocket',
    image: 'https://test.com',
    name: 'Rocket',
    reserved: false,
  };

  const tree = renderer.create(
    <Provider store={store}>
      <Rocket
        description={rocket.description}
        image={rocket.image}
        name={rocket.name}
        id={rocket.id}
        reserved={rocket.reserved}
      />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test if Rockets renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
