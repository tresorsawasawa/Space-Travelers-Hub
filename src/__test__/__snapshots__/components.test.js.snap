// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Rocket render correctly 1`] = `
<div
  className="rocket-container flex-center-center"
>
  <div
    className="image-container"
  >
    <img
      alt="Rocket"
      src="https://test.com"
    />
  </div>
  <div
    className="rocket-infos"
  >
    <h3
      className="rocket-name"
    >
      Rocket
    </h3>
    <p
      className="description"
    >
      This is the Falcon1 Rocket
    </p>
    <button
      className="reserve-btn clickable"
      id={1}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;

exports[`Test if Rockets List renders correctly 1`] = `
<section>
  <div
    className="rockets"
  >
    <ul
      className="rockets-list"
    />
  </div>
</section>
`;

exports[`Test if Rockets renders correctly 1`] = `
<section>
  <div
    className="rockets"
  >
    <ul
      className="rockets-list"
    />
  </div>
</section>
`;

exports[`Test if rockets render correctly 1`] = `
<div
  className="rocket-container flex-center-center"
>
  <div
    className="image-container"
  >
    <img
      alt="Rocket"
      src="https://test.com"
    />
  </div>
  <div
    className="rocket-infos"
  >
    <h3
      className="rocket-name"
    >
      Rocket
    </h3>
    <p
      className="description"
    >
      This is the Falcon1 Rocket
    </p>
    <button
      className="reserve-btn clickable"
      id={1}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
